cyber-security/modules/csls_s3_object_logging/cloudwatch.tf

```diff
@@ -0,0 +1,66 @@
+resource "aws_cloudwatch_event_rule" "s3_events" {
+  name        = local.cloudwatch_event_rule_name
+  description = "S3 object level logging rule"
+
+  tags = merge(var.tags, map("Name", local.cloudwatch_event_rule_name))
+
+  event_pattern = <<PATTERN
+  {
+  "source": [
+    "aws.s3"
+  ],
+  "detail-type": [
+    "AWS API Call via CloudTrail"
+  ],
+  "detail": {
+    "eventSource": [
+      "s3.amazonaws.com"
+    ],
+    "eventName": [
+      "ListObjects",
+      "ListObjectVersions",
+      "PutObject",
+      "GetObject",
+      "HeadObject",
+      "CopyObject",
+      "GetObjectAcl",
+      "PutObjectAcl",
+      "CreateMultipartUpload",
+      "ListParts",
+      "UploadPart",
+      "CompleteMultipartUpload",
+      "AbortMultipartUpload",
+      "UploadPartCopy",
+      "RestoreObject",
+      "DeleteObject",
+      "DeleteObjects",
+      "GetObjectTorrent",
+      "SelectObjectContent",
+      "PutObjectLockRetention",
+      "PutObjectLockLegalHold",
+      "GetObjectLockRetention",
+      "GetObjectLockLegalHold"
+    ]
+  }
+}
+PATTERN
+}
+
+resource "aws_cloudwatch_event_target" "cloudwatch_target" {
+  rule      = aws_cloudwatch_event_rule.s3_events.name
+  target_id = "S3EventsLogGroup"
+  arn       = substr(aws_cloudwatch_log_group.s3_events_log_group.arn, 0, length(aws_cloudwatch_log_group.s3_events_log_group.arn) - 2)
+}
+
+resource "aws_cloudwatch_log_group" "s3_events_log_group" {
+  name = local.cloudwatch_log_group_name
+
+  tags = merge(var.tags, map("Name", local.cloudwatch_log_group_name))
+}
+
+resource "aws_cloudwatch_log_subscription_filter" "log_subscription" {
+  name            = "log_subscription"
```

Comment:
This name could be more specific. It works while you have the context of this terraform but without that it is too generic.

Reply:
Done